lib/functions.php

```diff
@@ -752,9 +752,20 @@ function coroutine(callable $func) {
  * Upon resolution the Generator return value is used to succeed the promised result. If an
  * error occurs during coroutine resolution the returned promise fails.
  *
- * @param \Generator $generator The generator to resolve as a coroutine
+ * @param \Generator|callable $generator The generator to resolve as a coroutine
+ * @return \Amp\Promise
  */
-function resolve(\Generator $generator) {
+function resolve($generator) {
+    if (is_callable($generator)) {
```

Comment:
> technically, yes; but is there a reason to?

`Closure` should be faster.

> If you come that way, we just as well may reject the whole PR?!

No, accepting `Closure` provides value, because it provides some sugar, especially for PHP 5, where you can't use the universal variable syntax. Accepting other values however, doesn't provide any additional value?